ext/webstorage/02_indexeddb.js

```diff
@@ -0,0 +1,2454 @@
+  class IDBRequest extends EventTarget {
+    constructor() {
+      super();
+      webidl.illegalConstructor();
+    }
+
+    [_processed];
+    [_done] = false;
+
+    [_result];
+    // Ref: https://w3c.github.io/IndexedDB/#dom-idbrequest-result
+    get result() {
+      webidl.assertBranded(this, IDBRequestPrototype);
+      if (!this[_done]) {
+        throw new DOMException("", "InvalidStateError");
+      }
+      if (this[_error]) {
+        return undefined;
+      } else {
+        return this[_result];
+      }
+    }
+
+    [_error] = null;
+    get error() {
+      webidl.assertBranded(this, IDBRequestPrototype);
+      if (!this[_done]) {
+        throw new DOMException("", "InvalidStateError");
+      }
+      return this[_error];
+    }
+
+    [_source] = null;
+    get source() {
+      webidl.assertBranded(this, IDBRequestPrototype);
+      return this[_source];
+    }
+
+    [_transaction] = null;
+    get transaction() {
+      webidl.assertBranded(this, IDBRequestPrototype);
+      return this[_transaction];
+    }
+
+    get readyState() {
+      webidl.assertBranded(this, IDBRequestPrototype);
+      return this[_done] ? "done" : "pending";
+    }
+  }
+  defineEventHandler(IDBRequest.prototype, "success");
+  defineEventHandler(IDBRequest.prototype, "error");
+
+  webidl.configurePrototype(IDBRequest);
+  const IDBRequestPrototype = IDBRequest.prototype;
+
+  // Ref: https://w3c.github.io/IndexedDB/#idbopendbrequest
+  class IDBOpenDBRequest extends IDBRequest {
+    constructor() {
+      super();
+      webidl.illegalConstructor();
+    }
+  }
+  defineEventHandler(IDBOpenDBRequest.prototype, "blocked");
+  defineEventHandler(IDBOpenDBRequest.prototype, "upgradeneeded");
+
+  webidl.configurePrototype(IDBOpenDBRequest);
+
+  // Ref: https://w3c.github.io/IndexedDB/#open-a-database
+  function openDatabase(name, version) {
+    // TODO
+  }
+
+  // Ref: https://w3c.github.io/IndexedDB/#delete-a-database
+  function deleteDatabase(name, request) {
+    // TODO
+  }
+
+  // Ref: https://w3c.github.io/IndexedDB/#idbfactory
+  class IDBFactory {
+    constructor() {
+      webidl.illegalConstructor();
+    }
+
+    // Ref: https://w3c.github.io/IndexedDB/#dom-idbfactory-open
+    open(name, version = undefined) {
+      webidl.assertBranded(this, IDBFactoryPrototype);
+      const prefix = "Failed to execute 'open' on 'IDBFactory'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
+      name = webidl.converters.DOMString(name, {
+        prefix,
+        context: "Argument 1",
+      });
+      if (version !== undefined) {
+        version = webidl.converters["unsigned long long"](version, {
+          prefix,
+          context: "Argument 2",
+          enforceRange: true,
+        });
+      }
+
+      if (version === 0) {
+        throw new TypeError();
+      }
+
+      const request = webidl.createBranded(IDBOpenDBRequest);
+
+      (async () => {
```

Comment:
Perhaps my JS-fu is lacking but what's the point of this async IIFE? It's created, invoked, try-catched and finishes immediately, there is no await point so at least I cannot see anything to this except maybe strack trace differences here?

Reply:
its not done yet.